### DISCUSSION TITLE
Need to do destroy when using certain before hooks in rspec (i.e. before(:all))

### DISCUSSION THREAD

**Discussion by Em-Arce** [avionschool/batch8-activities#15] [Ideas]
Source: 

https://medium.com/swlh/deep-dive-into-database-transactions-in-rspec-a3d883b35c0
https://relishapp.com/rspec/rspec-rails/docs/transactions

**Comment by ianmandap** [comment]
Thanks Em!

I also found this [stackoveflow](https://stackoverflow.com/a/37761580/15233426) thread useful. In essence:

> `before :all` is **not** DB `transactional`. I.e, anything within the `before :all` persists to the DB and you must manually tear down in the `after :all` method.
>
> Implications mean that after test suites have completed, changes are not reverted ready for later tests. This can result in complicated bugs and issues with cross contamination of data. I.e, If an exception is thrown the `after :all` callback is not called.
>
> However, `before: each` **is** DB transaction[al].
